account_asset_management_no_depreciation/models/account_asset.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+READONLY_STATES = {
+    "open": [("readonly", True)],
+    "close": [("readonly", True)],
+    "removed": [("readonly", True)],
+}
+
+
+class AccountAsset(models.Model):
+    _inherit = "account.asset"
+
+    no_depreciation = fields.Boolean(
+        string="Non-Depreciation",
+        compute="_compute_no_depreciation",
+        readonly=False,
+        store=True,
+        states=READONLY_STATES,
+        help="Check this if you do not want to compute depreciation.",
+    )
+
+    @api.depends("profile_id")
+    def _compute_no_depreciation(self):
+        for asset in self:
+            asset.no_depreciation = asset.profile_id.no_depreciation
+
+    @api.onchange("no_depreciation")
+    def _onchange_no_depreciation(self):
+        if not self.no_depreciation:
+            self.method_number = 0
+            self.salvage_value = 0.0
+
+    @api.constrains("no_depreciation", "method_number")
+    def _check_method(self):
+        if any(a.no_depreciation and a.method_number != 0 for a in self):
+            raise UserError(
+                _("Non-Depreciation is only supported for Number of Years = 0.")
+            )
+
+    def remove(self):
```

Comment:
I wonder if this is correct, accounting wise. When no_depreciation = True, this doesn't nothing except change asset's state.

BUT,

If you look at normal asset profile (no_depreciation = False), while Depreciation is not running yet, it still product a Journal Entry for asset removal.

Following is what happened,

1. A removal line is created in Depreciation Table,
![Selection_095](https://user-images.githubusercontent.com/1973598/131311108-e04f96a3-21d1-46b6-9346-d1a87b6aadfd.png)

2. A journal entry is created for asset removal (don't worry about account code, I made it up).
![Selection_096](https://user-images.githubusercontent.com/1973598/131311208-c0f8c67c-189d-4d13-951f-cccca43cae52.png)

@peerayaproud, can you confirm this?

Reply:
For no_depreciation is true, it's doing nothing except changing state as you understand.
Also when we remove it from asset list, there shouldn't be any journal entry created since we've already recorded it as expense. Only state is changed from running > remove

For no_depreciation is false, 
After we've created asset but not running depreciation yet, the journal entry of asset removal should not be created yet unless we  remove it.